File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    remain = []
    task = 1

    for i in range(len(speeds)):
        if (100-progresses[i])%speeds[i] != 0:
            remain.append(int((100-progresses[i])/speeds[i]) + 1)
        else:
            remain.append(int((100-progresses[i])/speeds[i]))

    m = remain[0]
    for i in range(1, len(remain)):
        if remain[i] <= m:
            task+=1
        else:
            answer.append(task)
            task = 1
            m = remain[i]

    answer.append(task)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        int n = progresses.length;
        
        for (int i = 0; i < n; i++) {
            int remain = 100 - progresses[i];
            int day = remain / speeds[i];
            if(remain % speeds[i] != 0)
                day++;
            q.add(day);
        }
        
        int cnt = 1;
        int cur = q.poll();
        while(!q.isEmpty()){
            if(q.peek() <= cur){
                cnt++;
                q.poll();
            }
            else {
                answer.add(cnt);
                cnt = 1;
                cur = q.poll();
            }
        }
        answer.add(cnt);
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}